build/media_source/com_joomlaupdate/js/default.es6.js

```diff
@@ -11,13 +11,45 @@ Joomla = window.Joomla || {};
   Joomla.submitbuttonUpload = () => {
     const form = document.getElementById('uploadForm');
     const confirmBackup = document.getElementById('joomlaupdate-confirm-backup');
+    const file = form.install_package.files[0];
 
-    // do field validation
-    if (form.install_package.value === '') {
-      alert(Joomla.Text._('COM_INSTALLER_MSG_INSTALL_PLEASE_SELECT_A_PACKAGE'), true);
-    } else if (form.install_package.files[0].size > form.max_upload_size.value) {
-      alert(Joomla.Text._('COM_INSTALLER_MSG_WARNINGS_UPLOADFILETOOBIG'), true);
-    } else if (confirmBackup && confirmBackup.checked) {
+    // Make sure there is a package selected
+    if (!file) {
+      Joomla.renderMessages({ error: [Joomla.Text._('COM_INSTALLER_MSG_INSTALL_PLEASE_SELECT_A_PACKAGE')] });
+
+      return;
+    }
+
+    // Make sure this is a .zip package based on its file extension and on its content type
+    if (!file.name.match(/.zip$/i) || (file.type !== 'application/zip' && file.type !== 'application/x-zip-compressed')) {
+      Joomla.renderMessages({ error: [Joomla.Text._('COM_JOOMLAUPDATE_VIEW_UPLOAD_ERROR_NOTZIP')] });
+
+      return;
+    }
+
+    // Make sure this is not a Full Package based on its name
+    if (!form.install_package.value.match(/^Joomla_-(.*)-Full_Package.zip$/i)) {
```

Comment:
```suggestion
    if (form.install_package.value.match(/^Joomla_-(.*)-Full_Package.zip$/i)) {
```
From reading the code and the comment above it, I would expect here no negation of the match because the error shall be shown when the package indeed IS a full install package, and I see no negation in the regular expression for the match.

I haven't done a real test yet, so either I'm wrong and it works as expected, or previous tests were wrong.

Reply:
I real test shows it works as it should, but from reading the code I don't understand why.

Reply:
I actually believe something is actually amiss. It should be `if (form.install_package.value.match(/^Joomla_(.*)-Full_Package.zip$/i)) {` Note that I removed a dash. The filenames are like `Joomla_4.2.5-Stable-Full_Package.zip`.

Can you file a PR to my PR with this? I am currently working on something entirely different and I am in no good point to do a stash and swap branches.

BTW, this line can be read as follows: if the form.install_package.value string is not a case-independent string starting with `Joomla_` and ending with `-Full_Package.zip`. Given the way full packages are named, negating the match to their naming pattern will indeed return true for non-full packages, i.e. upgrade packages (as we only have those two types).

It's standard regex black magic f..kery :p

Reply:
Well, I understand regex syntax, that's why I was so confused. Will test and make PR for you when I have something working and I know why it does 😄

Reply:
The problem is that at least on Windows, the `form.install_package.value` contains also a fake path. I've verified that with an `alert(form.install_package.value)`. I think it should be enough to remove the start of string anchor `^` from the beginning of the regular expressions.

Reply:
Oh, and you should have escaped the dot of the extension at a few places.

Reply:
@nikosdion See https://github.com/nikosdion/joomla-cms/pull/13 .

Reply:
Of course I have tested my PR. The reason why the code here as it is seems to work was just lucky circumstances because the Regex after the one mentioned here was also wrong.